packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -749,6 +749,45 @@ void main() {
     },
   );
 
+  testWidgets(
+    'clear button tapped alse call onChanged when text not empty',
+        (WidgetTester tester) async {
+      String value = '';
+      final TextEditingController controller = TextEditingController();
+      await tester.pumpWidget(
+        CupertinoApp(
+            home: StatefulBuilder(
+                builder: (BuildContext context, StateSetter setState) {
+                  return Center(
+                    child: CupertinoTextField(
+                      controller: controller,
+                      placeholder: 'placeholder',
+                      onChanged: (String newValue) {
+                        setState(() {
+                          value = newValue;
+                        });
+                      },
+                      clearButtonMode: OverlayVisibilityMode.always,
+                    ),
+                  );
+                })
+        ),
+      );
+
+      expect(value, isEmpty);
+      controller.text = 'text entry';
+      await tester.pump();
+
+      await tester.tap(find.byIcon(CupertinoIcons.clear_thick_circled));
+      await tester.pump();
+
+      expect(controller.text, '');
+      expect(find.text('text entry'), findsNothing);
+      expect(value, 'text entry');
```

Comment:
Isn't this the opposite of what you want to test? That value is also now '' because the clear button called onChanged?

Reply:
yes, I make a mistake, it should be '' when clear button called the onChanged return.